wptherml/tests/test_exciton.py

```diff
@@ -0,0 +1,117 @@
+"""
+Unit and regression test for the wpspec package.
+"""
+
+# Import package, test suite, and other packages as needed
+import wptherml
+import numpy as np
+import pytest
+import sys
+
+
+""" perform a series of tests that would apply to the simple
+    case of a r = 100e-9 m spehere made of glass with ri = 1.5+0j
+    in air with ri = 1.0+0j
+"""
+sf = wptherml.SpectrumFactory()
+args = {
+'exciton_energy': 1.5,
+'aggregate_shape': (2,1,1),
+'displacement_vector' : [10000, 0, 0],  
+'transition_dipole_moment' : np.array([0, 0, 0.5]),
+'refractive_index' : 1.0,
+}
+
+exciton_test = sf.spectrum_factory('Frenkel', args)
+
+""" Define a second test instance that will
+    provide a simple case to test the dynamics... simple meaning
+    no coupling, so the transition_dipole_moment is set to zero
+    so that the V terms are 0 and so any localized exciton state
+    (i.e. c_vector = [1, 0] or c_vector = [0, 1] will be an eigenstate)
+    of the exciton Hamiltonian
+"""
+dynamics_args = {
+'exciton_energy': 1,
+'aggregate_shape': (2,1,1),
+'displacement_vector' : [1000000, 0, 0],  
+'transition_dipole_moment' : np.array([0, 0, 0.0]),
+'refractive_index' : 1.0,
+}
+
+dynamics_test = sf.spectrum_factory("Frenkel", dynamics_args)
+
+def test_compute_H0_element():
+
+    _H_11 = exciton_test._compute_H0_element(1, 1)
+    _H_12 = exciton_test._compute_H0_element(1, 2)
+
+    _expected_H_11 = exciton_test.exciton_energy
+    _expected_H_12 = 0.
+
+    assert np.isclose(_H_11, _expected_H_11, 1e-5)
+    assert np.isclose(_H_12, _expected_H_12, 1e-5)
+
+
+def test_compute_dipole_dipole_coupling():
+
+    _V_test = exciton_test._compute_dipole_dipole_coupling(0, 1)
+
+    _V_expected = 0
+
+    assert np.isclose(_V_test, _V_expected)
+
+
+
+def test_build_exciton_hamiltonian():
```

Comment:
@lanetolley This unit tests needs to be improved, it only tests a simple case where there are 2 monomers and no coupling.  Just like the dipole-dipole coupling, you need to test several non-trivial cases.

Since the dipole-dipole coupling unit test will catch any potential bugs in the dipole-dipole coupling function between any 2 monomers, the unit test for the build_exciton_hamiltonian method should focus on if the correct coupling is captured when there are more than 2 monomers... i.e. maybe a 3 monomer case testing coupling between units all in-plane and then a 4 monomer case testing out-of-plane coupling.